src/trie/path.rs

```diff
@@ -0,0 +1,126 @@
+use bitvec::{order::Msb0, vec::BitVec};
+use parity_scale_codec::{Decode, Encode, Error, Input, Output};
+
+use super::{merkle_node::Direction, TrieKey};
+
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Path(pub BitVec<u8, Msb0>);
+
+impl Encode for Path {
+    fn encode_to<T: Output + ?Sized>(&self, dest: &mut T) {
+        // Inspired from scale_bits crate but don't use it to avoid copy and u32 length encoding
```

Comment:
Can you explain the algo a little bit?
Give an example.

Reply:
Coz I get the idea that it's compressing the bits. But I don't understand how by reading this code

Reply:
I linked the original source code + added an explanation. Tell me you need anything more :)